src/rdf4cpp/datatypes/registry/util/ConstexprString.hpp

```diff
@@ -18,57 +18,63 @@ namespace rdf4cpp::datatypes::registry::util {
  */
 template<size_t N>
 struct ConstexprString {
-    std::array<char, N> value;
+    std::array<char, N> value{};
 
-    constexpr ConstexprString(const char (&str)[N]) noexcept {
+    constexpr ConstexprString() noexcept = default;
+
+    constexpr ConstexprString(char const (&str)[N]) noexcept {
         std::copy_n(str, N, value.data());
     }
+
     constexpr ConstexprString(std::string_view v) {
-        if (v.size()+1 != N)
-            throw std::invalid_argument{"size missmatch"};
-        std::copy_n(v.data(), N-1, value.data());
-        value[N-1] = 0;
+        if (v.size() != N - 1) {
+            throw std::invalid_argument{"size mismatch"};
+        }
+
+        std::copy_n(v.data(), v.size(), value.data());
```

Comment:
I would stick here with 
```suggestion
        std::copy_n(v.data(), N - 1, value.data());
```

since `v.size() == N - 1` is guranteed.